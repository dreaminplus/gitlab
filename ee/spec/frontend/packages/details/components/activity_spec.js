import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { GlSkeletonLoader } from '@gitlab/ui';
import PackageActivity from 'ee/packages/details/components/activity.vue';
import {
  npmPackage,
  mavenPackage as packageWithoutBuildInfo,
  mockPipelineInfo,
} from '../../mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('PackageActivity', () => {
  let wrapper;
  let store;

  function createComponent(
    packageEntity = packageWithoutBuildInfo,
    hasPipeline = false,
    isLoading = false,
    pipelineError = null,
  ) {
    store = new Vuex.Store({
      state: {
        isLoading,
        packageEntity,
        pipelineInfo: mockPipelineInfo,
        pipelineError,
      },
      getters: {
        packageHasPipeline: () => hasPipeline,
      },
    });

    wrapper = shallowMount(PackageActivity, {
      localVue,
      store,
    });
  }

  const commitInfo = () => wrapper.find({ ref: 'commit-info' });
  const pipelineInfo = () => wrapper.find({ ref: 'pipeline-info' });
  const pipelineLoadingSkeleton = () => wrapper.findAll(GlSkeletonLoader);
  const pipelineErrorMessage = () => wrapper.find({ ref: 'pipeline-error' });

  afterEach(() => {
    if (wrapper) wrapper.destroy();
  });

  describe('render', () => {
    it('to match the default snapshot when no pipeline', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });

    it('to match the default snapshot when there is a pipeline', () => {
      createComponent(npmPackage, true, false, 'pipeline error');

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('pipeline information', () => {
    it('does not display pipeline information when no build info is available', () => {
      createComponent();

      expect(pipelineInfo().exists()).toBe(false);
    });

    it('displays the loading skeleton when fetching information', () => {
      createComponent(npmPackage, true, true);

      expect(pipelineLoadingSkeleton()).toHaveLength(2);
    });

    it('displays that the pipeline error information fetching fails', () => {
      const pipelineError = 'an-error-message';
      createComponent(npmPackage, true, false, pipelineError);

      expect(commitInfo().exists()).toBe(true);
      expect(pipelineLoadingSkeleton().exists()).toBe(false);
      expect(pipelineErrorMessage().exists()).toBe(true);
      expect(pipelineErrorMessage().text()).toBe(pipelineError);
    });

    it('displays the pipeline information if found', () => {
      createComponent(npmPackage, true);

      expect(commitInfo().exists()).toBe(true);
      expect(pipelineInfo().exists()).toBe(true);
    });
  });
});
