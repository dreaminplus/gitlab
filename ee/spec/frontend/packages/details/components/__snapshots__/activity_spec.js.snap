// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageActivity render to match the default snapshot when no pipeline 1`] = `
<div
  class="append-bottom-default"
>
  <h3
    class="gl-font-size-16"
  >
    Activity
  </h3>
   
  <div
    class="info-well"
  >
    <!---->
     
    <!---->
     
    <div
      class="well-segment d-flex align-items-center"
    >
      <gl-icon-stub
        class="append-right-8 d-none d-sm-block"
        name="clock"
        size="16"
      />
       
      <gl-sprintf-stub
        message="Published to the repository at %{timestamp}"
      />
    </div>
  </div>
</div>
`;

exports[`PackageActivity render to match the default snapshot when there is a pipeline 1`] = `
<div
  class="append-bottom-default"
>
  <h3
    class="gl-font-size-16"
  >
    Activity
  </h3>
   
  <div
    class="info-well"
  >
    <div
      class="well-segment"
    >
      <span>
        pipeline error
      </span>
    </div>
     
    <!---->
     
    <div
      class="well-segment d-flex align-items-center"
    >
      <gl-icon-stub
        class="append-right-8 d-none d-sm-block"
        name="clock"
        size="16"
      />
       
      <gl-sprintf-stub
        message="Published to the repository at %{timestamp}"
      />
    </div>
  </div>
</div>
`;
